mne/channels/tests/test_channels.py

```diff
@@ -360,4 +361,58 @@ def test_equalize_channels():
     assert epochs is epochs2
 
 
+def test_combine_channels():
+    """Test channel combination on Raw, Epochs, and Evoked."""
+    raw = read_raw_fif(raw_fname, preload=True)
+    raw_ch_bad = read_raw_fif(raw_fname, preload=True)
+    raw_ch_bad.info['bads'] = ['MEG 0113', 'MEG 0112']
+    epochs = Epochs(raw, read_events(eve_fname), preload=True)
+    evoked = epochs.average()
+    good = dict(foo=[0, 1, 3, 4], bar=[5, 2])  # good grad and mag
+
+    # Test good cases
+    combine_channels(raw, good)
+    combine_channels(epochs, good)
+    combine_channels(evoked, good)
+    combine_channels(raw, good, keep_stim=True)
+    combine_channels(raw, good, drop_bad=True)
+    combine_channels(raw_ch_bad, good, drop_bad=True)
+
+    # Test result with one ROI
+    good_single = dict(foo=[0, 1, 3, 4])  # good grad
+    combined_mean = combine_channels(raw, good_single, method='mean')._data
+    combined_median = combine_channels(raw, good_single, method='median')._data
+    combined_std = combine_channels(raw, good_single, method='std')._data
+    foo_mean = np.mean(raw._data[good_single['foo']], axis=0)
+    foo_median = np.median(raw._data[good_single['foo']], axis=0)
+    foo_std = np.std(raw._data[good_single['foo']], axis=0)
+    assert np.array_equal(combined_mean, np.expand_dims(foo_mean, axis=0))
+    assert np.array_equal(combined_median, np.expand_dims(foo_median, axis=0))
+    assert np.array_equal(combined_std, np.expand_dims(foo_std, axis=0))
+
+    # Test bad cases
+    raw_no_preload = read_raw_fif(raw_fname, preload=False)
+    raw_no_stim = read_raw_fif(raw_fname, preload=True)
+    raw_no_stim.pick_types(meg=True, stim=False)
+    bad1 = dict(foo=[0, 376], bar=[5, 2])  # out of bounds
+    bad2 = dict(foo=[0, 2], bar=[5, 2])  # type mix in same group
+    pytest.raises(RuntimeError, combine_channels, raw_no_preload, good)
+    pytest.raises(ValueError, combine_channels, raw, good, method='bad_method')
```

Comment:
In new or updated tests, we are migrating to using `pytest.raises` as a context manager. Example:

```suggestion
    with pytest.raises(ValueError, match='"method" must be a callable, or '):
        combine_channels(raw, good, method='bad_method') 
```

Where `match` is a regex that will match the actual error message that is raised.